microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/openapi/properties/MicroProfileOpenAPIProvider.java

```diff
@@ -0,0 +1,112 @@
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.openapi.properties;
+
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.OPEN_API_CONFIG;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.MODEL_READER;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.FILTER;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCAN_DISABLE;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCAN_PACKAGES;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCAN_CLASSES;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCAN_EXCLUDE_PACKAGES;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCAN_EXCLUDE_CLASSES;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SERVERS;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SERVERS_PATH;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SERVERS_OPERATION;
+import static com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants.SCHEMA;
+
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+
+import com.redhat.microprofile.jdt.core.AbstractStaticPropertiesProvider;
+import com.redhat.microprofile.jdt.core.IPropertiesCollector;
+import com.redhat.microprofile.jdt.core.SearchContext;
+import com.redhat.microprofile.jdt.core.utils.JDTTypeUtils;
+
+/**
+ * Properties provider that provides static MicroProfile OpenAPI properties
+ * 
+ * @author David Kwon
+ * 
+ * @see https://github.com/eclipse/microprofile-open-api/blob/master/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#311-core-configurations
+ *
+ */
+public class MicroProfileOpenAPIProvider extends AbstractStaticPropertiesProvider {
+
+	@Override
+	protected boolean isAdaptedFor(SearchContext context, IProgressMonitor monitor) {
+		IJavaProject javaProject = context.getJavaProject();
+		return (JDTTypeUtils.findType(javaProject, OPEN_API_CONFIG) != null);
+	}
+
+	@Override
+	protected void collectStaticProperties(SearchContext context, IProgressMonitor monitor) {
+		IPropertiesCollector collector = context.getCollector();
+		String docs = "Configuration property to specify the fully qualified name of the OASModelReader implementation.";
+		super.addItemMetadata(collector, MODEL_READER, "java.util.Optional<java.lang.String>", docs, null, null,
```

Comment:
To know the type of each openapi properties, I think we should see in smallrye implementation. For instance https://github.com/smallrye/smallrye-open-api/blob/51e103fd5c2dcdc83136f45798330903aecbb9de/implementation/src/main/java/io/smallrye/openapi/api/OpenApiConfigImpl.java#L74 you can see that model reader is a String and not an java.util.Optional<java.lang.String>. So I think the correct type is `java.lang.String`. It's the same thing for the other properties.

Reply:
Sounds good. Just double checking, properties like `mp.openapi.scan.packages`: https://github.com/smallrye/smallrye-open-api/blob/51e103fd5c2dcdc83136f45798330903aecbb9de/implementation/src/main/java/io/smallrye/openapi/api/OpenApiConfigImpl.java#L107

have a `Set<String>` type. Should I go ahead with `java.lang.String` since we do not have validation support for `Set<String>`? Or maybe the type should be `java.util.Set<java.lang.String>` in case we do support that type in the future?

Reply:
fixed